File: 2021 maj/python/4.py
with open("instrukcje.txt", "r") as f:
    inst = f.read().strip().split('\n')

ans, alf = [], "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
for line in inst:
    s, c = line.split()
    if s == "DOPISZ":
        ans.append(c)
    elif s == "ZMIEN":
        ans[-1] = c
    elif s == "USUN":
        ans.pop()
    elif s == "PRZESUN":
        try:
            idx = ans.index(c)
        except ValueError:
            continue

        if c == 'Z':
            ans[idx] = 'A'
        else:
            ans[idx] = alf[alf.find(c) + 1]
print(''.join(ans))

File: 2021 maj/c++/4.cpp
#include <iostream>
#include <fstream>
#include <string>
using namespace std;


int main()
{
    ifstream in("instrukcje.txt");
    
    string s, ans;
    char c;
    while (in >> s >> c)
    {
        if (s == "DOPISZ")
            ans.push_back(c);
        else if (s == "ZMIEN")
            ans[ans.size() - 1] = c;
        else if (s == "USUN")
            ans.pop_back();
        else if (s == "PRZESUN")
        {
            unsigned long long idx = ans.find(c);
            if (idx == string::npos)
                continue;
            if (c == 'Z')
                ans[idx] = 'A';
            else
                ans[idx]++;
        }
    }
    cout << ans << '\n';
}